lib/puppet/util/platform.rb

```diff
@@ -33,6 +33,22 @@ def fips_enabled?
       def self.jruby?
         RUBY_PLATFORM == 'java'
       end
+
+      def jruby_fips?
+        @@jruby_fips ||= if RUBY_PLATFORM == 'java'
+                           require 'java'
+
+                           begin
+                             Java::OrgBouncycastleCrypto::AsymmetricKeyPair
+                             false
+                           rescue NameError
+                             true  # The class is only in non-FIPS BouncyCastle
+                           end
+                         else
+                           false
+                         end
+      end
+      module_function :jruby_fips?
```

Comment:
Is this check sensible?

Reply:
Seems simple enough, if there isn't a more obvious check (like a class that is only in FIPS and is named accordingly). But it seems like generally speaking the FIPS stuff is a subset of regular BC?

Reply:
That's been my impression. The lack of this class is the first thing you see when trying to load jopenssl in FIPS.